docs/for-developers/technical-overview.md

```diff
@@ -3,71 +3,200 @@ id: technical-overview
+
+In our example **sender A** wants to stream **5** per cycle.
+
+If the sender would start/end exactly at the beginning of a cycle we would have a 
+- start delta (+5)
+- end delta (-5)
+
+Which we would store on the timeline of the receiver.
+
+### Calculation of Deltas
+Normally, a Drip will not start exactly at the beginning of a cycle. A stream will start most of the time during an ongoing cycle.
+
+Therefore, we need to split the start and end deltas into two seperate deltas.
+
+Let’s take a look at the example from **sender A**, how to calculate the deltas.
+
+![](https://i.imgur.com/BoDNwLq.png)
+
+**Deltas: Start Sending**
+The sender will be sending 1 per second or 5 per cycle. The senders usually are sending constant per-cycle amounts over long periods of time, so the added values tend to create long series of constant numbers, in this case, 5s.
+
+We exploit this observation to turn the per-cycle flows of funds into **deltas** relative to the previous cycles. Now that we need to store data only for cycles where the funding rate changes, it's very cheap. This is what the contract actually stores: a mapping from cycle numbers to deltas.
+
+
+In the case of **sender 1** the total change of the per-cycle delta is +5 to start sending.
+The current cycle isn't fully affected though, only 3 out of 5 seconds are sending.
+It's effectively going to transfer only the amount of 3, which is reflected in the +3 delta change.
+On the other hand, the next cycle and the ones after it are going to transfer the full 5.
+This is expressed with the +2 delta change, which turns 2 per cycle into the full 5 per cycle
+
+When funding is stopped, the deltas need to be reverted. To do that, essentially the same process is applied, just with negative deltas.
+In this case, the reverting is split into 2 cycles too, one with -4 and the other with -1.
+
+## Storing Deltas
+For are complete example, we can see the delta calculation for both **sender 1** and **sender 2**.
+
+From the perspective of the receiver each cycle has one delta value. This delta value is modified if sender start/stops a stream in this speicific cycle.
+
+![](https://i.imgur.com/RxRvxfY.png)
+
+
+The stored delta for a cycle in the timeline can be seen as the sum of individuals sender deltas for that cycle.
+
+**Actual Storage In The Contract**
+![](https://i.imgur.com/oh4wuRo.png)
+
+Here we can see to represent the two senders in the timeline of the receiver, we only modified 5 different cycle deltas.
+
+
+## Collecting
+
+The receiver can at any time collect the funds sent to it. The contract calculates the total amount and then transfers it out to the receiver's wallet. The collected amount is always everything available at a given moment, there's no way to limit it.
```

Comment:
That's true that you can't receive a fraction of a cycle, but you can limit the number of analyzed cycles. We're also about to merge squeezing which further splits up the current cycle into receiving from particular users. I'm not sure if it's worth noting here considering the noise it brings.

Reply:
Yes -- lots of tradeoffs around trying to keep things simple in this page. I think we'll go into squeezing in more detail in a separate docs page and then add a link to that here. I've tried to clean up this paragraph a bit for now.